api/app/Models/User.php

```diff
@@ -207,7 +208,13 @@ public static function scopeIsProfileComplete(Builder $query, ?bool $isProfileCo
                 $query->orWhereNotNull('looking_for_bilingual');
             });
             $query->whereNotNull('is_gov_employee');
-            $query->whereNotNull('has_priority_entitlement');
+            $query->where(function (Builder $query) {
+                $query->whereNotNull('has_priority_entitlement')
+                    ->orWhere(function (Builder $query) {
+                        $query->where('has_priority_entitlement', true)
+                            ->whereNotNull('priority_number');
+                    });
+            });
```

Comment:
https://recordit.co/0crnaSC0G0
I'm not 100% sure on this bit 

`$query->whereNotNull('has_priority_entitlement')` 
will immediately return people with it as true but the number as null right?

Maybe instead something like
```
                $query->where('has_priority_entitlement', false)
                    ->orWhere(function (Builder $query) {
                        $query->where('has_priority_entitlement', true)
                            ->whereNotNull('priority_number');
```

Reply:
You are completely right, thanks! Fixed in [fix profile complete scope](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7313/commits/426adc12dafb520421d265601f96ade9d3ace2ad)